crates/bevy_math/src/cubic_splines.rs

```diff
@@ -846,11 +830,287 @@ impl<P: Point> IntoIterator for CubicCurve<P> {
     }
 }
 
+impl<P: Point> RationalGenerator<P> for CubicNurbs<P> {
+    #[inline]
+    fn to_curve(&self) -> RationalCurve<P> {
+        let segments = self
+            .control_points.windows(4)
+            .zip(self.weights.windows(4))
+            .zip(self.knot_vector.windows(8))
+            .map(|((points, weights), knot_vector_segment)| {
+                let knot_vector_segment = knot_vector_segment
+                    .try_into()
+                    .expect("Knot vector windows are of length 8");
+                let matrix = Self::generate_matrix(knot_vector_segment);
+                RationalSegment::coefficients(
+                    points
+                        .try_into()
+                        .expect("Points vector windows are of length 4"),
+                    weights
+                        .try_into()
+                        .expect("Weights vector windows are of length 4"),
+                    matrix,
+                )
+            })
+            .collect();
+        RationalCurve { segments }
+    }
+}
+
+/// Implement this on cubic splines that can generate a curve from their spline parameters.
+pub trait RationalGenerator<P: Point> {
+    /// Build a [`RationalCurve`] by computing the interpolation coefficients for each curve segment.
+    fn to_curve(&self) -> RationalCurve<P>;
+}
+
+/// A segment of a rational cubic curve, used to hold precomputed coefficients for fast interpolation.
+///
+/// Segments can be chained together to form a longer compound curve.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct RationalSegment<P: Point> {
```

Comment:
We should probably also add a `span: Range<f32>` here derived from the knot value, then use it to look up the appropriate segment.